src/core/consent/createConsent.js

```diff
@@ -80,7 +80,13 @@ export default ({
     // lifecycle.onResponse notify consent of a consent response so that
     // the consent state can be unsuspended and sendEdgeNetworkRequest
     // can finish its process.
-    consentRequestComplete: consentState.unsuspend,
+    consentRequestComplete() {
+      if (queue.length === 0) {
+        // If we have more setConsent requests that need to run, run them
+        // before unsuspending the consent.
+        consentState.unsuspend();
+      }
+    },
```

Comment:
Thanks for doing this. For some background: I had this at one point in time but removed it because of, if I remember correct, a circular dependency issue (a component's lifecycle method was waiting for consent and consent was waiting for the component's lifecycle method to finish). Waiting for all consent responses before unsuspending wasn't deemed critical (I remember discussing it with Justin, Joe, or Jon). When we switched to calling `consentRequestComplete` from the privacy component's `onResponse` lifecycle, I believe that eliminated the circular dependency issue we had, so I think this should work fine now and is preferable. 👍